### ISSUE TITLE
DPO reports training-set preference accuracy as if it were held out

### ISSUE THREAD

**Issue by metzgerdev** [metzgerdev/midi_gpt#3]
`finetune_dpo.py:242-243` builds the evaluation loader from the same pairs it trains on:

```python
loader      = to_loader(pairs, args.batch_size, shuffle=True)
eval_loader = to_loader(pairs, args.batch_size, shuffle=False)
```

So the line printed at the end —

```
[bass] reward margin +0.000 -> +2.431   preference accuracy 48% -> 97%
```

— is a fit statistic, not a generalization one. It measures how well the policy memorized the pairs it just trained on, and will approach 100% regardless of whether anything useful was learned.

## Why it matters more here than usual

Each distinct edit is augmented into 12 pairs (one per key, `finetune_dpo.py:119`), and the default is `--epochs 15`. A handful of edited sections therefore gets ~180 effective passes over the same information. Overfitting is the expected outcome, and the reported accuracy is exactly the number that cannot reveal it.

The `beta` anchor to the frozen reference and the base-corpus regression check are the only real guards, and the regression check silently disappears when `training_data/` is absent (`base_corpus_loader` returns `None`, which is the documented behaviour since the corpus is not tracked in git).

## Fix

Hold out a fraction of the pairs, grouped by `(run, section)` so all 12 transpositions of a section stay on one side — the same grouping problem as #2. Report accuracy on the held-out pairs, and keep the training-set number only if it is labelled as such.

With very few pairs, leave-one-section-out would be more honest than a fixed split.
